cmd/mdatagen/templates/readme.md.tmpl

```diff
@@ -16,6 +17,9 @@
 {{- if .Status.Warnings }}
 | Warnings      | [{{ stringsJoin .Status.Warnings ", " }}](#warnings) |
 {{- end }}
+{{- if ne $class "" }}
+| Issues        | ![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3A{{ $class }}%2F{{ $type }}%20&label=open&color=orange&logo=opentelemetry) ![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3A{{ $class }}%2F{{ $type }}%20&label=closed&color=blue&logo=opentelemetry) |
```

Comment:
Label names are created from the directory names by trimming the `class` suffix while the `type` in metadata.yaml is the name of the component in the user configuration. So this link will not catch issues for the components where those two values are different, e.g., `splunk_hec`/`splunkhec`. 

We need to either update the label names to use metadata.yaml's `type` or use the directory name here similar to how it's done in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/loader.go#L233

Reply:
I would update labels, I think we have a few cases but nothing too hard.

Reply:
I'm good with that, it just can be tricky. The bash CI automation would need to read metadata.yaml files

Reply:
Understood, I didn’t know they were generated

Reply:
Slept on it, and you're right. That's way too much drama, I'll use the directory name same as we did before. We will need to address this eventually maybe.

Reply:
> We will need to address this eventually maybe.

Right, having the `type` in the GH label eventually is preferable. Worth filing an issue

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24171